docs/jenkins.md

```diff
@@ -0,0 +1,190 @@
+* create a foreman_example.yaml file in [foreman-infra/JJB](https://github.com/theforeman/foreman-infra/tree/master/puppet/modules/jenkins_job_builder/files/theforeman.org/yaml/jobs/plugins)
+  * This will create a "test_plugin_foreman_example_master" job in Jenkins to test the master branch.
+* ensure the job is green by fixing bugs, installing dependencies etc.
+* add hook to GitHub repo, see [GitHub repo hook](https://projects.theforeman.org/projects/foreman/wiki/Jenkins#Testing-develop)
+
+An org admin must then:
+
+* add the repo to the [Bots team](https://github.com/orgs/theforeman/teams/bots/repositories) with **write** access
+
+### Smart proxy plugin testing
+
+Proxy plugins are tested like ordinary gems with tests run entirely from the plugin directory, installing the smart proxy as a dependency (via bundler's git support).  The [test_proxy_plugin_matrix job](https://ci.theforeman.org/job/test_proxy_plugin_matrix/) is usually used to test plugins in a generic way.
+
+Each plugin should have a job defined in JJB that calls test_proxy_plugin_matrix here: https://ci.theforeman.org/view/Plugins/
+
+#### Smart proxy plugin PR testing
+
+To test pull requests, a separate job is used that also takes the PR details: https://ci.theforeman.org/view/Plugins/job/test_proxy_plugin_pull_request/
+
+### Adding a new smart proxy plugin
+
+For a plugin "smart_proxy_example", first create a job that tests the main (master or develop) branch.
+
+* ensure plugin tests run when doing `bundle install` and `rake test`, see [testing a plugin](https://projects.theforeman.org/projects/foreman/wiki/How_to_Create_a_Smart-Proxy_Plugin#Testing) for help
+* if different branches rely on different versions of smart proxy, specify `:branch` in Gemfile on those branches
+* create a smart_proxy_example.yaml file in foreman-infra/JJB
+* https://github.com/theforeman/foreman-infra/tree/master/puppet/modules/jenkins_job_builder/files/theforeman.org/yaml/jobs/plugins
+* This will create a "test_proxy_plugin_smart_proxy_example_master" job in Jenkins to test the master branch.
+* ensure the job is green by fixing bugs, installing dependencies etc.
+* add hook to GitHub repo, see [GitHub repo hook](https://projects.theforeman.org/projects/foreman/wiki/Jenkins#Testing-develop)
+
+Next, set up PR testing for the plugin:
+
+* add template to foreman-infra/puppet/modules/slave/templates, update branch, job names etc.
+
+An org admin must then:
+
+* add the repo to the [Bots team](https://github.com/orgs/theforeman/teams/bots/repositories) with **write** access
+
+## Job configurations
+
+### Testing develop
+
+All repos with an associated job that tests their master/develop branch should have a hook added to the repo to trigger immediate builds.
+
+To set up the hook, an org/repo admin must:
+
+* View the repository settings
+* Click _Webhooks
+  * _Add service_ dropdown, select _Jenkins (Git plugin)_
```

Comment:
This no longer works since GH doesn't have services anymore. It's replaced with:
* Payload URL: https://ci.theforeman.org/github-webhook/
* Content Type: application/json (default)
* Secret: the secret (which I think we haven't stored in a shared secret)
* Just the push event